homeassistant/components/pglab/__init__.py

```diff
@@ -0,0 +1,72 @@
+"""PG LAB Electronics integration."""
+
+from __future__ import annotations
+
+from pypglab.mqtt import (
+    Client as PyPGLabMqttClient,
+    Sub_State as PyPGLabSubState,
+    Subcribe_CallBack as PyPGLaSubscribeCallBack,
+)
+
+from homeassistant.components import mqtt
+from homeassistant.components.mqtt import ReceiveMessage
+from homeassistant.components.mqtt.subscription import (
+    async_prepare_subscribe_topics,
+    async_subscribe_topics,
+    async_unsubscribe_topics,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+
+from .discovery import PGLabDiscovery, create_discovery
+
+# Define custom PGLab Config entry.
+type PGLABConfigEntry = ConfigEntry[PGLabDiscovery]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: PGLABConfigEntry) -> bool:
+    """Set up PG LAB Electronics integration from a config entry."""
+
+    # Define the call back for PGLab module to publish a MQTT message.
+    async def mqtt_publish(topic: str, payload: str, qos: int, retain: bool) -> None:
+        await mqtt.async_publish(hass, topic, payload, qos, retain)
+
+    # Define the call back for PGLab module to subscribe to a MQTT message.
+    async def mqtt_subscribe(
+        sub_state: PyPGLabSubState, topic: str, callback_func: PyPGLaSubscribeCallBack
```

Comment:
Would it be possible to fix the spelling in the library?
```suggestion
        sub_state: PyPGLabSubState, topic: str, callback_func: PyPGLabSubscribeCallBack
```

Reply:
Thank you for spotting that... sure. I will fix it.